axelor-stock/src/main/java/com/axelor/apps/stock/web/StockMoveController.java

```diff
@@ -177,18 +172,57 @@ public void draft(ActionRequest request, ActionResponse response) {
 
   public void cancel(ActionRequest request, ActionResponse response) {
     try {
-      StockMove stockMove = request.getContext().asType(StockMove.class);
-
+      Context context = request.getContext();
+      StockMove stockMove = context.asType(StockMove.class);
+      Optional<Boolean> supplierArrivalCancellationAutomaticMail =
+          Optional.of(context)
+              .map(ctx -> ctx.get("_supplierArrivalCancellationAutomaticMail"))
+              .map(Boolean.class::cast);
+      Optional<Integer> supplierArrivalCancellationMessageTemplateID =
+          Optional.of(context)
+              .map(ctx -> ctx.get("_supplierArrivalCancellationMessageTemplate"))
+              .map(hash -> ((LinkedHashMap) hash).get("id"))
+              .map(Integer.class::cast);
```

Comment:
What is happening if _supplierArrivalCancellationAutomaticMail or _supplierArrivalCancellationAutomaticMail are null ?